UiTalkAsyncHandler.cs

```diff
@@ -0,0 +1,398 @@
+      Task.Run(() =>
+        {
+          try
+          {
+            TalkMessage talkMessage = this.FormatTalkMessage(nameToTranslate, textToTranslate);
+            TalkMessage foundTalkMessage = this.FindAndReturnTalkMessage(talkMessage);
+
+            if (foundTalkMessage != null)
+            {
+              this.translatedName = foundTalkMessage.TranslatedSenderName;
+              this.translatedText = foundTalkMessage.TranslatedTalkMessage;
+
+              PluginLog.Debug($"From database - Name: {foundTalkMessage.TranslatedSenderName}, Message: {foundTalkMessage.TranslatedTalkMessage}");
+            }
+            else
+            {
+              string textTranslation = this.Translate(textToTranslate);
+              string nameTranslation = nameToTranslate.IsNullOrEmpty() ? string.Empty : this.Translate(nameToTranslate);
+
+              this.translatedName = nameTranslation;
+              this.translatedText = textTranslation;
+
+              TalkMessage translatedTalkData = new TalkMessage(
+                 nameToTranslate,
+                 textToTranslate,
+                 ClientState.ClientLanguage.Humanize(),
+                 ClientState.ClientLanguage.Humanize(),
+                 nameTranslation,
+                 textTranslation,
+                 langDict[languageInt].Code,
+                 this.configuration.ChosenTransEngine,
+                 DateTime.Now,
+                 DateTime.Now);
+              string result = InsertTalkData(translatedTalkData);
+              PluginLog.Debug($"Talk Message DB Insert operation result: {result}");
+            }
+          }
+          catch (Exception e)
+          {
+            PluginLog.Warning("TranslateTalkReplacing Exception: " + e);
+          }
+        });
+    }
+
+    private unsafe void TranslateTalkReplacing()
+    {
+      PluginLog.Debug($"TranslateTalkReplacing");
+
+      if (this.configuration.UseImGuiForTalk && !this.configuration.SwapTextsUsingImGui)
+      {
+        return;
+      }
+
+      try
+      {
+        var addon = GameGui.GetAddonByName("Talk");
+        var talkAddon = (AtkUnitBase*)addon;
+        if (talkAddon == null || !talkAddon->IsVisible)
+        {
+          return;
+        }
+
+        var nameNode = talkAddon->GetTextNodeById(2);
+        var textNode = talkAddon->GetTextNodeById(3);
+        if (textNode == null || textNode->NodeText.IsEmpty)
+        {
+          return;
+        }
+
+        if (this.configuration.TranslateNpcNames && nameNode != null && !nameNode->NodeText.IsEmpty)
+        {
+          nameNode->SetText(this.translatedName);
+        }
+
+        var parentNode = talkAddon->GetNodeById(10);
+        textNode->TextFlags = (byte)(TextFlags)((byte)TextFlags.WordWrap | (byte)TextFlags.MultiLine);
+        textNode->SetWidth(parentNode->GetWidth());
+        textNode->SetText(this.translatedText);
+        textNode->ResizeNodeForCurrentText();
```

Comment:
using the width of the parent the resizing works

Reply:
Nice find!!